### PR TITLE
fix learning rate display in trainer when using galore optimizer

### LINKED ISSUE #30082: Galore ignores provided learning rate
### System Info

- `transformers` version: 4.39.3
- Platform: Linux-6.8.0-76060800daily20240311-generic-x86_64-with-glibc2.35
- Python version: 3.10.13
- Huggingface_hub version: 0.20.3
- Safetensors version: 0.4.2
- Accelerate version: 0.26.1
- Accelerate config: 	not found
- PyTorch version (GPU?): 2.2.0 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: True
- Using distributed or parallel set-up in script?: False

### Who can help?

@younesbelkada

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

I'm trying to finetune (extended with new tokens) Mistral 7B with Galore, but no matter where I pass the learning rate, trainer is using 0.001.

model setup:
```python
# source model is a path to Mistral 7B with some extra indices in embedding matrix.
model = MistralForCausalLM.from_pretrained(source_model,
                                           torch_dtype=torch.bfloat16,
                                           attn_implementation='flash_attention_2')

# I don't think it is important, but I will leave it here just in case:
# freeze embeddings that do not need gradient, train only new tokens
input_emb = model.get_input_embeddings()
input_emb.weight.register_hook(lambda grad: zero_grad(grad, list(range(0, 3200))))

model.set_input_embeddings(input_emb)

training_args = TrainingArguments(output_dir=output_model,
                                  evaluation_strategy="steps", warmup_ratio=0.05,
                                  overwrite_output_dir=True, gradient_accumulation_steps=1,
                                  per_device_train_batch_size=4, per_device_eval_batch_size=1,
                                  num_train_epochs=3, bf16_full_eval=True, bf16=True,
                                  eval_steps=1000, save_strategy='steps', save_steps=1000,
                                  save_total_limit=5, dataloader_num_workers=8,
                                  load_best_model_at_end=True,
                                  neftune_noise_alpha=0, 
                                  prediction_loss_only=True,
                                  gradient_checkpointing=True,
                                  optim="galore_adamw_8bit_layerwise",
                                  optim_target_modules=[
                                      'q_proj', 'k_proj', 'down_proj', 'up_proj',
                                      'gate_proj', 'v_proj', 'o_proj', 'lm_head'
                                  ],
                                  optim_args="""rank=512, update_proj_gap=500, scale=0.25, lr=0.0002""",
                                  learning_rate=0.0002,
                                  )

trainer = Trainer(
    model=model,
    args=training_args,
    eval_dataset=validation,
    train_dataset=dataset,
    data_collator=identity_collator)
```

Yet, stdout logs shows: {'loss': 4.6255, 'grad_norm': 0.0, 'learning_rate': 0.001, 'epoch': 0.01}, while I'm passing learning rate to both directly TrainingArguments and Galore's optim_args.

### Expected behavior

Learning rate shall be the same as passed to TrainingArguments via learning_rate argument.

### PR DESCRIPTION
**Pull request by vasqu**
# What does this PR do?
The new Galore optimizer uses a dummy optimizer which displays the wrong learning rate in the trainer. Since the real galore optimizers are applied parameter-wise for the groups each time before, I think it's just a display issue. 

Fixes #30082 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@younesbelkada 
@muellerzr

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks for adding - looks good to me cc @younesbelkada to confirm!

The failing errors are unrelated to this PR. Once a fix is merged into main, we'll let you know, you can rebase and then we can merge this in

**Review by younesbelkada — APPROVED** [review]
Makes sense, thanks !
